h2o-py/tests/testdir_algos/glm/pyunit_PUBDEV_7481_lambda_alpha_array_multinomial_coldStart.py

```diff
@@ -0,0 +1,63 @@
+import sys
+sys.path.insert(1,"../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.glm import H2OGeneralizedLinearEstimator as glm
+
+# Given arrays of alpha and no lambda, the lambdas will be estimated as the max of gradient divided by
+# the corresponding alpha value.  
+#
+# When an array of alpha and/or lambdas are given, a list of submodels are also built.  For each submodel built, only
+# the coefficients, lambda/alpha/deviance values are returned.  The model metrics is calculated from the submodel
+# with the best deviance.  
+#
+# In this test, in addition, we build separate models using just one lambda and one alpha values as when building one
+# submodel.  In theory, the coefficients obtained from the separate models should equal to the submodels.  We check 
+# and compare the followings:
+# 1. coefficients from submodels and individual model should match when they are using the same alpha/lambda value;
+# 2. training metrics from alpha array should equal to the individual model matching the alpha/lambda value;
+def glm_alpha_array_lambda_null():
+    # first test: compare coefficients and deviance
+    keySets = ["MSE", "null_deviance", "logloss", "RMSE", "r2"]
+    d = h2o.import_file(path=pyunit_utils.locate("smalldata/covtype/covtype.20k.data"))
+    mL = glm(family='multinomial',alpha=[0.1,0.5,0.9], Lambda=[0.1, 0.5, 0.9], cold_start=True)
+    d[54] = d[54].asfactor()
+    mL.train(training_frame=d,x=list(range(0,54)),y=54)
+    r = glm.getGLMRegularizationPath(mL)
+    regKeys = ["alphas", "lambdas", "explained_deviance_valid", "explained_deviance_train"]
+    best_submodel_index = mL._model_json["output"]["best_submodel_index"]
+    coefClassSet = ['coefs_class_0','coefs_class_1', 'coefs_class_2', 'coefs_class_3', 'coefs_class_4','coefs_class_5', 
+                    'coefs_class_6', 'coefs_class_7']
+    coefClassSetNorm = ['std_coefs_class_0','std_coefs_class_1', 'std_coefs_class_2', 'std_coefs_class_3', 
+                        'std_coefs_class_4', 'std_coefs_class_5', 'std_coefs_class_6', 'std_coefs_class_7']
+    startVal = [0]*371
+    startVal[52] = -1.9166426513747452
+    startVal[105] = -1.200312843494449
+    startVal[158] = -2.2256240518579173
+    startVal[211] = -2.2256240518579173
+    startVal[264] = -2.121931593300788
+    startVal[317] = -2.2256240518579173
+    startVal[370] = -2.2256240518579173
```

Comment:
is it possible to write tests without those magic numbers?

Reply:
Challenge accepted and completed.